extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/group/UserGroupService.java

```diff
@@ -208,13 +208,21 @@ private ExprNode getGroupSearchFilter() throws GuacamoleException {
             }
         }
 
+        // Gather all attributes relevant for a group
+        ArrayList<String> groupAttributes = new ArrayList<String>();
```

Comment:
1. The size of the required backing array can be anticipated here. This will work as-is, but unnecessarily relies on automatic array resizing or sufficiently-large defaults.
2. Unless code consuming this relies on the object being an `ArrayList`, the type declared for the variable should be kept minimal, presumably `List<String>` or `Collection<String>`.